### ISSUE TITLE
Can't Enable Payment Plugins

### ISSUE THREAD

**Issue by skillsgarage** [pretix/pretix#1399]
Hi, I understand that this request isn't of a total technical nature for which I apologise but I was just looking for some guidance.

I have installed pretix on our server and have it all up and running. This was installed via Docker.

When we try to activate any of the installed plugins (or deactivate any that are enabled as part of the install) we are met with a server error.

Please, could you help me to understand where the log files would be that would show what causes this error? Or, indeed if there are any other steps that we should take before activating the plugins in the installed plugin list?

Thank you in advance for any assistance.

David

**Comment by raphaelm** [comment]
The logs would be stored within the data volume that you probably mounted into your docker container. (And yes, the docs would be required to say more about this.)

**Comment by skillsgarage** [comment]
Not at all. You do cleatly state that you need some experience of using Docker etc. It's our first attempt at using it so it's a bit of a learning expereince.

We installed from your instructions so will have a dig around to see if we can locate them and figure this out.

I am right in thinking that they are already installed and just need activating?

Thank you for your very quick response too.

David

**Comment by raphaelm** [comment]
If you follow the tutorial closely, the log files will be in a subdirectory of /var/pretix-data.

**Comment by skillsgarage** [comment]
Thanks. Found them. The log is showing the following. It seems to relate to the file directory python3.6. We only have python 2.7 and 3.7 in there. Is there something that I need to change is it likely the error isn't related to this?

`ERROR 2019-09-11 13:40:37,968 django.request log Internal Server Error: /control/event/revlearning/tmmk/settings/plugins
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/site-packages/kombu/utils/functional.py", line 42, in __call__
    return self.__value__
AttributeError: 'ChannelPromise' object has no attribute '__value__'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.6/site-packages/kombu/transport/virtual/base.py", line 921, in create_channel
    return self._avail_channels.pop()
IndexError: pop from empty list

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.6/site-packages/kombu/utils/objects.py", line 42, in __get__
    return obj.__dict__[self.__name__]
KeyError: 'client'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.6/site-packages/kombu/transport/redis.py", line 874, in _prepare_virtual_host
    vhost = int(vhost)
ValueError: invalid literal for int() with base 10: '2eventr'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.6/site-packages/django/core/handlers/exception.py", line 34, in inner
    response = get_response(request)
  File "/usr/local/lib/python3.6/site-packages/django/core/handlers/base.py", line 115, in _get_response
    response = self.process_exception_by_middleware(e, request)
  File "/usr/local/lib/python3.6/site-packages/django/core/handlers/base.py", line 113, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/pretix/src/pretix/control/permissions.py", line 27, in wrapper
    return function(request, *args, **kw)
  File "/usr/local/lib/python3.6/site-packages/django/views/generic/base.py", line 71, in view
    return self.dispatch(request, *args, **kwargs)
  File "/usr/local/lib/python3.6/site-packages/django/views/generic/base.py", line 97, in dispatch
    return handler(request, *args, **kwargs)
  File "/pretix/src/pretix/control/views/event.py", line 252, in post
    self.object.enable_plugin(module, allow_restricted=allow_restricted)
  File "/pretix/src/pretix/base/models/event.py", line 835, in enable_plugin
    regenerate_css.apply_async(args=(self.pk,))
  File "/usr/local/lib/python3.6/site-packages/celery/app/task.py", line 570, in apply_async
    **options
  File "/usr/local/lib/python3.6/site-packages/celery/app/base.py", line 756, in send_task
    amqp.send_task_message(P, name, message, **options)
  File "/usr/local/lib/python3.6/site-packages/celery/app/amqp.py", line 552, in send_task_message
    **properties
  File "/usr/local/lib/python3.6/site-packages/kombu/messaging.py", line 181, in publish
    exchange_name, declare,
  File "/usr/local/lib/python3.6/site-packages/kombu/connection.py", line 510, in _ensured
    return fun(*args, **kwargs)
  File "/usr/local/lib/python3.6/site-packages/kombu/messaging.py", line 187, in _publish
    channel = self.channel
  File "/usr/local/lib/python3.6/site-packages/kombu/messaging.py", line 209, in _get_channel
    channel = self._channel = channel()
  File "/usr/local/lib/python3.6/site-packages/kombu/utils/functional.py", line 44, in __call__
    value = self.__value__ = self.__contract__()
  File "/usr/local/lib/python3.6/site-packages/kombu/messaging.py", line 224, in <lambda>
    channel = ChannelPromise(lambda: connection.default_channel)
  File "/usr/local/lib/python3.6/site-packages/kombu/connection.py", line 852, in default_channel
    self.ensure_connection(**conn_opts)
  File "/usr/local/lib/python3.6/site-packages/kombu/connection.py", line 422, in ensure_connection
    callback, timeout=timeout)
  File "/usr/local/lib/python3.6/site-packages/kombu/utils/functional.py", line 341, in retry_over_time
    return fun(*args, **kwargs)
  File "/usr/local/lib/python3.6/site-packages/kombu/connection.py", line 275, in connect
    return self.connection
  File "/usr/local/lib/python3.6/site-packages/kombu/connection.py", line 823, in connection
    self._connection = self._establish_connection()
  File "/usr/local/lib/python3.6/site-packages/kombu/connection.py", line 778, in _establish_connection
    conn = self.transport.establish_connection()
  File "/usr/local/lib/python3.6/site-packages/kombu/transport/virtual/base.py", line 941, in establish_connection
    self._avail_channels.append(self.create_channel(self))
  File "/usr/local/lib/python3.6/site-packages/kombu/transport/virtual/base.py", line 923, in create_channel
    channel = self.Channel(connection)
  File "/usr/local/lib/python3.6/site-packages/kombu/transport/redis.py", line 513, in __init__
    self.client.ping()
  File "/usr/local/lib/python3.6/site-packages/kombu/utils/objects.py", line 44, in __get__
    value = obj.__dict__[self.__name__] = self.__get(obj)
  File "/usr/local/lib/python3.6/site-packages/kombu/transport/redis.py", line 982, in client
    return self._create_client(asynchronous=True)
  File "/usr/local/lib/python3.6/site-packages/kombu/transport/redis.py", line 945, in _create_client
    return self.Client(connection_pool=self.async_pool)
  File "/usr/local/lib/python3.6/site-packages/kombu/transport/redis.py", line 976, in async_pool
    self._async_pool = self._get_pool(asynchronous=True)
  File "/usr/local/lib/python3.6/site-packages/kombu/transport/redis.py", line 949, in _get_pool
    params = self._connparams(asynchronous=asynchronous)
  File "/usr/local/lib/python3.6/site-packages/kombu/transport/redis.py", line 924, in _connparams
    connparams.pop('virtual_host', None))
  File "/usr/local/lib/python3.6/site-packages/kombu/transport/redis.py", line 878, in _prepare_virtual_host
    vhost,
ValueError: Database is int between 0 and limit - 1, not 2eventr`

**Comment by raphaelm** [comment]
It looks like your configuration is wront, ``2eventr`` is not a valid redis database

**Comment by skillsgarage** [comment]
How stupid of me. I must have left that file open while typing something else and it's ended up in there. I didn't even spot that as an issue in the log. It works perfectly now.

Thank you so much for your help with this and your super quick responses.

**Comment by raphaelm** [comment]
Great!

**Comment by skillsgarage** [comment]
Can I bug you with one more question? Is it possible to change the text in the application, for example via translation?

I saw in the docs about using 

[languages]
allow_incubating=pt-br,da

[languages]
path=/path/to/my/translations

So we created a new .po file and added it to a directory and added the path and the 2 letter code but the text doesn't change. Again, just our lack of knowledge about Docker (we couldn't get the manual installation to work).

Alternatively, in a docker install, where would the language files (infact all of the files that would be under the src directory be held)?

Thanks again

David
